build_all: display: Fails on boards already declaring display support

**Describe the bug**

#79545 extended `build_all/display` coverage to more boards, specially ones already declaring `display support`.
Test is now built on boards with existing display support using the overlay file provided for that test which enable more display support:
https://github.com/zephyrproject-rtos/zephyr/blob/5be4aed6209963cb26ee5c6171a595c2440f03fd/tests/drivers/build_all/display/app.overlay#L37-L45

It now happens that 2 different versions of displays are being compiled on one single board, as this is the case for `stm32f429i_disc1` which compiles with `"ilitek,ili9342c"` and `"ilitek,ili9341"` enabled, leading to a conflict in headers:

```
In file included from /local/home/zephyrproject/zephyr/drivers/display/display_ili9xxx.c:563:
/local/home/zephyrproject/zephyr/drivers/display/display_ili9342c.h:70:43: error: conflicting types for 'ili9xxx_regs_0'; have 'const struct ili9342c_regs'
   70 |         static const struct ili9342c_regs ili9xxx_regs_##n = {          \
      |                                           ^~~~~~~~~~~~~
[...]
In file included from /local/home/zephyrproject/zephyr/drivers/display/display_ili9xxx.c:558:
/local/home/zephyrproject/zephyr/drivers/display/display_ili9341.h:124:42: note: previous definition of 'ili9xxx_regs_0' with type 'const struct ili9341_regs'
  124 |         static const struct ili9341_regs ili9xxx_regs_##n = {                                      \
      |                                          ^~~~~~~~~~~~~
[...]
```

**To Reproduce**
1. west twister -p stm32f429i_disc1 -s tests/drivers/build_all/display/drivers.display.build.default
2. See error

**Expected behavior**

Test should compile.
I'm not sure the issue actually comes from the test definition. One can argue that one board should support using these two dfifferent versions of displays.

**Impact**

CI issue

**Environment (please complete the following information):**

 - Commit SHA or Version used: v4.0.0-978-g113875d53c7

--- a/drivers/display/display_ili9340.h
+++ b/drivers/display/display_ili9340.h
@@ -50,7 +50,7 @@ struct ili9340_regs {
 
 /* Initializer macro for ILI9340 registers. */
 #define ILI9340_REGS_INIT(n)                                                   \
-	static const struct ili9340_regs ili9xxx_regs_##n = {                  \
+	static const struct ili9340_regs ili9340_regs_##n = {                  \
 		.gamset = DT_PROP(DT_INST(n, ilitek_ili9340), gamset),         \
 		.frmctr1 = DT_PROP(DT_INST(n, ilitek_ili9340), frmctr1),       \
 		.disctrl = DT_PROP(DT_INST(n, ilitek_ili9340), disctrl),       \

--- a/drivers/display/display_ili9341.h
+++ b/drivers/display/display_ili9341.h
@@ -121,7 +121,7 @@ struct ili9341_regs {
 		     "ili9341: Error length frame rate control (IFCTL) register");                 \
 	BUILD_ASSERT(DT_PROP_LEN(DT_INST(n, ilitek_ili9341), etmod) == ILI9341_ETMOD_LEN,          \
 		     "ili9341: Error length entry Mode Set (ETMOD) register");                     \
-	static const struct ili9341_regs ili9xxx_regs_##n = {                                      \
+	static const struct ili9341_regs ili9341_regs_##n = {                                      \
 		.gamset = DT_PROP(DT_INST(n, ilitek_ili9341), gamset),                             \
 		.ifmode = DT_PROP(DT_INST(n, ilitek_ili9341), ifmode),                             \
 		.frmctr1 = DT_PROP(DT_INST(n, ilitek_ili9341), frmctr1),                           \

--- a/drivers/display/display_ili9342c.h
+++ b/drivers/display/display_ili9342c.h
@@ -67,7 +67,7 @@ struct ili9342c_regs {
 
 /* Initializer macro for ILI9342C registers. */
 #define ILI9342c_REGS_INIT(n)                                           \
-	static const struct ili9342c_regs ili9xxx_regs_##n = {		\
+	static const struct ili9342c_regs ili9342c_regs_##n = {		\
 	.gamset = DT_PROP(DT_INST(n, ilitek_ili9342c), gamset),		\
 	.ifmode = DT_PROP(DT_INST(n, ilitek_ili9342c), ifmode),		\
 	.frmctr1 = DT_PROP(DT_INST(n, ilitek_ili9342c), frmctr1),	\

--- a/drivers/display/display_ili9488.h
+++ b/drivers/display/display_ili9488.h
@@ -44,7 +44,7 @@ struct ili9488_regs {
 
 /* Initializer macro for ILI9488 registers. */
 #define ILI9488_REGS_INIT(n)                                                   \
-	static const struct ili9488_regs ili9xxx_regs_##n = {                  \
+	static const struct ili9488_regs ili9488_regs_##n = {                  \
 		.frmctr1 = DT_PROP(DT_INST(n, ilitek_ili9488), frmctr1),       \
 		.disctrl = DT_PROP(DT_INST(n, ilitek_ili9488), disctrl),       \
 		.pwctrl1 = DT_PROP(DT_INST(n, ilitek_ili9488), pwctrl1),       \

--- a/drivers/display/display_ili9xxx.c
+++ b/drivers/display/display_ili9xxx.c
@@ -517,7 +517,7 @@ static const struct ili9xxx_quirks ili9488_quirks = {
 #define ILI9XXX_INIT(n, t)                                                     \
 	ILI##t##_REGS_INIT(n);                                                 \
 									       \
-	static const struct ili9xxx_config ili9xxx_config_##n = {              \
+	static const struct ili9xxx_config ili9##t##_config_##n = {            \
 		.quirks = &ili##t##_quirks,                                    \
 		.mipi_dev = DEVICE_DT_GET(DT_PARENT(INST_DT_ILI9XXX(n, t))),   \
 		.dbi_config = {                                                \
@@ -535,15 +535,15 @@ static const struct ili9xxx_quirks ili9488_quirks = {
 		.x_resolution = ILI##t##_X_RES,                                \
 		.y_resolution = ILI##t##_Y_RES,                                \
 		.inversion = DT_PROP(INST_DT_ILI9XXX(n, t), display_inversion),\
-		.regs = &ili9xxx_regs_##n,                                     \
+		.regs = &ili##t##_regs_##n,                                    \
 		.regs_init_fn = ili##t##_regs_init,                            \
 	};                                                                     \
 									       \
-	static struct ili9xxx_data ili9xxx_data_##n;                           \
+	static struct ili9xxx_data ili9##t##_data_##n;                         \
 									       \
 	DEVICE_DT_DEFINE(INST_DT_ILI9XXX(n, t), ili9xxx_init,                  \
-			    NULL, &ili9xxx_data_##n,                           \
-			    &ili9xxx_config_##n, POST_KERNEL,                  \
+			    NULL, &ili9##t##_data_##n,                         \
+			    &ili9##t##_config_##n, POST_KERNEL,                \
 			    CONFIG_DISPLAY_INIT_PRIORITY, &ili9xxx_api)
 
 #define DT_INST_FOREACH_ILI9XXX_STATUS_OKAY(t)                                 \

